pkg/app/pipedv1/eventwatcher/eventwatcher.go

```diff
@@ -748,17 +757,30 @@ func modifyText(path, regexText, newValue string) ([]byte, bool, error) {
 	if firstGroup == "" {
 		return nil, false, fmt.Errorf("capturing group not found in the given regex")
 	}
-	subRegex, err := pool.Get(firstGroup)
-	if err != nil {
-		return nil, false, fmt.Errorf("failed to compile the first capturing group: %w", err)
-	}
 
 	var touched, outDated bool
 	newText := regex.ReplaceAllFunc(content, func(match []byte) []byte {
 		touched = true
-		outDated = string(subRegex.Find(match)) != newValue
-		// Return text replacing the only first capturing group with the newValue.
-		return subRegex.ReplaceAll(match, []byte(newValue))
+		submatches := regex.FindSubmatchIndex(match)
+
+		if len(submatches) < 4 {
+			return match
+		}
+
+		groupStart, groupEnd := submatches[2], submatches[3]
+		if string(match[groupStart:groupEnd]) != newValue {
+			outDated = true
+		} else {
+			return match
+		}
```

Comment:
[nit] prefer to the early return to minify the scope of the variable
```suggestion
                // no update on the value
		if string(match[groupStart:groupEnd]) == newValue {
			return match
		}
		
		outDated = true
```